server/routes/applications.ts

```diff
@@ -0,0 +1,442 @@
+
+    if(!req.session.user)
+        return res.status(401).json({ message: 'User is not authorized, must be logged in.' });
+
+    if(!(await isUsersApplication(req.session.user._id, applicationId)))
+        return res.status(401).json({ message: 'requested application is not attached to the user.' });
+
+    if(!req.body)
+        return res.status(400).json({ message: 'no body provided for event creation.' });
+
+    let title: string, date: string, location: string;
+    try {
+        ({ title, date, location } = req.body);
+        if(!title || !checkNonEmptyString(title))
+            throw new Error('Event Title must be provided.');
+        if(!date || !checkDate(date))
+            throw new Error('Event Date must be provided.');
+        if(!location || !checkNonEmptyString(location))
+            throw new Error('Event Location must be provided.');
+    } catch(e) {
+        return res.status(400).json({ message: e.message });
+    }
+
+    try {
+        const event: EventObject = await createEvent(req.session.user._id, applicationId, title, date, location);
+        res.json(event);
+    } catch(e) {
+        res.status(500).json({ message: 'problem creating event for the given application.', error: e.message });
+    }
+
+});
+
+// POST /contact/:applicationId
+router.post('/contact/:applicationId', async (req, res) => {
+    const { applicationId } = req.params;
+    if(!applicationId || !checkNonEmptyString(applicationId))
+        return res.status(400).json({ message: 'application id must be a string representing a mongoDB ObjectId.' });
+
+    if(!req.session.user)
+        return res.status(401).json({ message: 'User is not authorized, must be logged in.' });
+
+    if(!(await isUsersApplication(req.session.user._id, applicationId)))
+        return res.status(401).json({ message: 'requested application is not attached to the user.' });
+
+    if(!req.body)
+        return res.status(400).json({ message: 'no body provided for contact creation.' });
+
+    let name: string, pronouns: string, location: string, phone: string, email: string;
+    try {
+        ({ name, pronouns, location, phone, email } = req.body);
+        if(!name || !checkNonEmptyString(name))
+            throw new Error('Contact Name must be provided.');
+        if(!pronouns || !checkNonEmptyString(pronouns))
+            throw new Error('Contact Pronouns must be provided.');
+        if(!location || !checkNonEmptyString(location))
+            throw new Error('Contact Location must be provided.');
+        if(!phone || !checkNonEmptyString(phone))
+            throw new Error('Contact Phone must be provided.');
+        if(!email || !checkNonEmptyString(email))
+            throw new Error('Contact Email must be provided.');
+    } catch(e) {
+        return res.status(400).json({ message: e.message });
+    }
+
+    try {
+        // TODO: use data functions for adding contact to application
+        const contact: ContactObject = null;
```

Comment:
Mike's code for this is labeled in the other pull request if you wanna fill it in here preemptively

Reply:
can't use his code until it is merged into main, but will only be like 1 additional line of code here